TheKeyOAuth2/TheKeyOAuth2LoginViewController.m

```diff
@@ -77,8 +77,32 @@ - (void)configureNavButtons {
     self.navigationItem.rightBarButtonItems = @[self.forwardBarButtonItem, self.backBarButtonItem];
 }
 
+/* defaults to true if locale or language code cannot be determined */
 -(bool)isLeftToRightLanguage {
-    return ![@"ar" isEqualToString:[[NSLocale currentLocale] languageCode]];
+    NSLocale *locale = [NSLocale currentLocale];
+    if (!locale) {
+        return true;
+    }
+    
+    NSString *languageCode;
+    
+    if (@available(iOS 10, *)) {
+        languageCode = [locale languageCode];
+    } else {
+        NSString *language = [[NSLocale preferredLanguages] objectAtIndex:0];
+        NSDictionary *languageDic = [NSLocale componentsFromLocaleIdentifier:language];
+        languageCode = [languageDic objectForKey:@"kCFLocaleLanguageCodeKey"];
+    }
+    
+    if (!languageCode) {
+        return true;
+    }
+    
+    if (![languageCode respondsToSelector:@selector(isEqualToString:)]) {
+        return true;
+    }
+    
+    return ![languageCode isEqualToString:@"ar"];
```

Comment:
Do you mean for this to be "en" or "en_US"?  Just wondering.

Reply:
Nope. It's meant to be `ar`. Arabic is the only app interface that we currently actively support that is right to left. All the others are left to right.